src/qt/intro.cpp

```diff
@@ -251,24 +251,29 @@ void Intro::setStatus(int status, const QString &message, quint64 bytesAvailable
     {
         ui->freeSpace->setText("");
     } else {
-        QString freeString = tr("%n GB of free space available", "", bytesAvailable/GB_BYTES);
-        if (bytesAvailable < m_required_space_gb * GB_BYTES) {
-            freeString += " " + tr("(of %n GB needed)", "", m_required_space_gb);
-            ui->freeSpace->setStyleSheet("QLabel { color: #800000 }");
-            ui->prune->setChecked(true);
-        } else if (bytesAvailable / GB_BYTES - m_required_space_gb < 10) {
-            freeString += " " + tr("(%n GB needed for full chain)", "", m_required_space_gb);
-            ui->freeSpace->setStyleSheet("QLabel { color: #999900 }");
-            ui->prune->setChecked(true);
```

Comment:
> The difference with this PR is just that the checkbox gets unchecked when a new datadir is selected, where previously it would never get unchecked?

Steps to reproduce:
1) the default datadir not having enough free space causes the prune checkbox gets checked (both on master and with this PR)
2) a user chooses a custom datadir with enough free space:
  - on master the prune checkbox remains checked
  - with this PR the prune checkbox gets unchecked

Reply:
In commit "gui: Add Intro::UpdateFreeSpaceLabel()" (a208b68b6f30f932d82e0d625bb230f9c1d5ab72)

> > The difference with this PR is just that the checkbox gets unchecked when a new datadir is selected, where previously it would never get unchecked?
> 
> Steps to reproduce: [...]

Thanks for the description. It sounds like the answer is yes.